src/interface/report/PlayerLoader.tsx

```diff
@@ -62,148 +59,188 @@ interface State {
   combatantsFightId: number | null;
 }
 
+interface StateFC extends State {
+  tanks: number;
+  healers: number;
+  dps: number;
+  ranged: number;
+  ilvl: number;
+}
+
 const defaultState: State = {
   error: null,
   combatants: null,
   combatantsFightId: null,
 };
 
-class PlayerLoader extends React.PureComponent<Props, State> {
-  tanks = 0;
-  healers = 0;
-  dps = 0;
-  ranged = 0;
-  ilvl = 0;
-
-  static getDerivedStateFromProps(props: Props, state: State) {
-    if (props.fight.id !== state.combatantsFightId) {
-      // When switching fights we need to unset combatants before rendering to avoid children from doing API calls twice
-      return defaultState;
+const defaultStateFC: StateFC = {
+  ...defaultState,
+  tanks: 0,
+  healers: 0,
+  dps: 0,
+  ranged: 0,
+  ilvl: 0,
+};
+
+type Action =
+  | { type: 'reset' }
+  | {
+      type: 'update_composition';
+      tanks: number;
+      healers: number;
+      dps: number;
+      ranged: number;
+      ilvl: number;
     }
-    return state;
+  | { type: 'set_combatants'; combatants: CombatantInfoEvent[]; combatantsFightId: number }
+  | { type: 'set_error'; error: Error };
+
+const fcReducer = (state: StateFC, action: Action): StateFC => {
+  switch (action.type) {
+    case 'reset':
+      return { ...defaultStateFC };
+    case 'update_composition':
+      return {
+        ...state,
+        tanks: action.tanks,
+        healers: action.healers,
+        dps: action.dps,
+        ranged: action.ranged,
+        ilvl: action.ilvl,
+      };
+    case 'set_combatants':
+      return {
+        ...state,
+        combatants: action.combatants,
+        combatantsFightId: action.combatantsFightId,
+      };
+    case 'set_error':
+      return { ...defaultStateFC, error: action.error };
+    default:
+      return { ...state };
   }
+};
 
-  state = defaultState;
-
-  componentDidMount() {
-    // noinspection JSIgnoredPromiseFromCall
-    this.loadCombatants(this.props.report, this.props.fight);
-    this.scrollToTop();
-  }
+const PlayerLoaderFC = ({ children, setCombatants, history, playerName, playerId }: Props) => {
+  const [
+    { error, combatants, combatantsFightId, tanks, healers, dps, ranged, ilvl },
+    dispatchFC,
+  ] = useReducer(fcReducer, defaultStateFC);
```

Comment:
we do have redux in the codebase, but i don't think that is an adventure for today :upside_down_face: